### PR TITLE
Make changes to annotation targets describing paths concluding with a navigation property

### PR DESCRIPTION
**Pull request by millicentachieng**
This PR:
- Remove markdown links from property descriptions to add as annotations
- Update regex to ensure paths with typecasts are properly added as annotation targets
-  Replaces line breaks with space in property descriptions
- Correctly append API version to link annotations
- Ensure functions/actions with parameters are properly identified and annotated
- Change how paths concluding with a navigation property are added as annotation targets e.g. now `microsoft.graph.GraphService/users/manager`, before `microsoft.graph.user/manager`.

### CODE REVIEW

File: ApiDoctor.Publishing/CSDL/csdlwriter.cs [hunk 1]
```diff
@@ -1654,13 +1647,15 @@ private static ActionOrFunctionBase FindActionOrFunctionTarget(string path, Enti
                     if (requestTarget.QualifiedType.IsCollection())
                         bindingParameterType = $"Collection({bindingParameterType})";
 
+                    int bracketStartIndex = requestParts.Last().IndexOf('(');
+                    var actionOrFunctionName = bracketStartIndex == -1 ? requestParts.Last() : requestParts.Last()[..bracketStartIndex];
```
Comment (andrueastman) [inline ApiDoctor.Publishing/CSDL/csdlwriter.cs:1651]:
```suggestion
                    int bracketStartIndex = requestParts[^1].IndexOf('(');
                    var actionOrFunctionName = bracketStartIndex == -1 ? requestParts[^1] : requestParts[^1][..bracketStartIndex];      
```

File: ApiDoctor.Publishing/CSDL/csdlwriter.cs [hunk 2]
```diff
@@ -638,29 +638,22 @@ private static void MergeAnnotations(string fullName, IODataAnnotatable annotata
...
                     {
-                        var existingAnnotation = annotations.AnnotationList.FirstOrDefault(a => a.Term == annotation.Term);
+                        var existingAnnotation = targetAnnotations.AnnotationList.FirstOrDefault(a => a.Term == annotation.Term);
                         if (existingAnnotation != null && annotation.Collection != null && annotation.Collection.Records?.Count > 0)
                         {
-                            if (existingAnnotation.Collection == null)
-                            {
-                                existingAnnotation.Collection = new RecordCollection();
-                            }
+                            existingAnnotation.Collection ??= new RecordCollection();
```
Comment (andrueastman) [inline ApiDoctor.Publishing/CSDL/csdlwriter.cs:648]: 👍🏼

**Review by andrueastman — COMMENTED** [review]
Any chance there's a diff of the metadata generated with changes in the PR?

**Comment by millicentachieng** [comment]
@andrueastman See how the metadata will look like [here](https://[raw.githubusercontent.com/microsoftgraph/msgraph-metadata/new-annotations-target-path/clean_v10_metadata/cleanMetadataWithDescriptionsAndAnnotationsv1.0.xml](https://raw.githubusercontent.com/microsoftgraph/msgraph-metadata/new-annotations-target-path/clean_v10_metadata/cleanMetadataWithDescriptionsAndAnnotationsv1.0.xml)).

**Review by andrueastman — APPROVED** [review]

**Pull request merged**